File: src/algorithms/is-prefix-of-word.js
/**
 * Given a sentence that consists of some words separated by a single space,
 * as well as a searchWord, returns the index (1-indexed) of the first word
 * that searchWord is a prefix of, or returns -1 if no such word exists.
 * 
 * @param {string} sentence - A string parameter made up of English words
 * @param {string} searchWord - A string of English characters
 * @returns {number} - Returns the index of the first matching word that
 *     searchWord is a prefix of (1-indexed)
 */
function isPrefixOfWord (sentence, searchWord) {
  let res;
  sentence = sentence.split(' ');
  for (const word of sentence) {
    if (word.slice(0, searchWord.length) === searchWord) {
      res = sentence.indexOf(word) + 1;
      break;
    }
    res = -1;
  }
  return res;
}

File: src/algorithms/is-prefix-of-word.py
def isPrefixOfWord(sentence, searchWord):
    """Given a sentence that consists of some words separated by a single 
    space, and a searchWord, check if searchWord is a prefix of any word in 
    sentence. Return the index of the word in sentence (1-indexed) where 
    searchWord is a prefix of this word. If searchWord is a prefix of more than 
    one word, return the index of the first word (minimum index). If there is 
    no such word return -1. A prefix of a string s is any leading contiguous 
    substring of s.

    Args:
        sentence (str): A string of lowercase English letters and spaces.
        searchWord (str): A string of lowercase English letters.

    Returns:
        int: Returns the index of the first word in sentence that is a prefix
        of searchWord.
    """
    sentence = sentence.split(" ")
    length = len(searchWord)
    for i in range(len(sentence)):
        if sentence[i][:length] == searchWord:
            return i + 1
    return -1
